scripts/im_plot.sl

```diff
@@ -33,38 +32,57 @@ start_time=`date +${runtime_fmt}`
 echo ___im plot___
 
 python $gmsim/workflow/scripts/cybershake/add_to_mgmt_queue.py $MGMT_DB_LOC/mgmt_db_queue $SRF_NAME IM_plot running
-res=`python $gmsim/visualization/im_plotting/im_plot.py $CSV_PATH $RRUP_OR_STATION_PATH --output $OUTPUT_XYZ_DIR`
+res=`python $gmsim/visualization/im_plotting/im_plot.py $CSV_PATH $STATION_FILE_PATH --output $OUTPUT_XYZ_DIR`
 
 exit_val=$?
 
-if [[ $exit_val == 0 ]]; then
-
-    echo "finished making csvs for im_plot"
-    ## Load python2 for plot_stations
+load_python2_mahuika () {
+    # Load python2, have to do this as virtualenv points to this python
+    # verions, which is not accessible without loading
     module load Python/2.7.14-gimkl-2017a
-    ## Reset the PYTHONPATH
+
+    # Reset the PYTHONPATH
     export PYTHONPATH=''
-    ## PYTHONPATH (this can be removed once qcore is installed as a pip package)
+
+    # PYTHONPATH (this can be removed once qcore is installed as a pip package)
     export PYTHONPATH=/nesi/project/nesi00213/opt/mahuika/qcore:$PYTHONPATH
-    ## PYTHONPATH for workflow
+
+    # PYTHONPATH for workflow
     export PYTHONPATH=/nesi/project/nesi00213/workflow:$PYTHONPATH
-    ## Load the python2 virtual environment
+
+    # Load the virtual environment
     source /nesi/project/nesi00213/share/virt_envs/python2_mahuika/bin/activate
+}
+
+plot_stations () {
+    for f in $OUTPUT_XYZ_DIR/*; do 
```

Comment:
my recommendation would be still have the for loop outside of the function just the other stuff here.